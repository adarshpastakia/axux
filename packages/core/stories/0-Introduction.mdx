import { Title, Canvas, Story } from "@storybook/addon-docs";
import { Anchor, Subheading } from "@storybook/blocks";
import { Meta } from "@storybook/blocks";

<Meta title="@core/Introduction" />

# Core Components

<Anchor storyId="install">
  <Subheading>Installing</Subheading>
</Anchor>

```bash

#npm
npm install @axux/core

# yarn
yarn add @axux/core

```

<Anchor storyId="using">
  <Subheading>Using AxUX</Subheading>
</Anchor>

```tsx
import { AxApplicationProvider,AxViewport } from "@axux/core";
import { I18nextProvider } from "react-i18next";
import { BrowserRouter } from "react-router";

export const Application = () => {
  return (
    {/** i18n for localization **/}
    <I18nextProvider>
      {/** react router **/}
      <BrowserRouter>
        <AxApplicationProvider>
          <AxViewport>
            .
            .
            .
            .
          </AxViewport>
        </AxApplicationProvider>
      </BrowserRouter>
    </I18nextProvider>
  );
};
```
