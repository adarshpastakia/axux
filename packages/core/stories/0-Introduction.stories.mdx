import { Title, Canvas, Story } from "@storybook/addon-docs";

<Meta title="@core/Introduction" />

<Title>Core Components</Title>
