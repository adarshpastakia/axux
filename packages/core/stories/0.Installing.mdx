import { Title, Canvas, Story } from "@storybook/addon-docs";
import { Anchor, Subheading } from "@storybook/blocks";
import { Meta } from "@storybook/blocks";

<Meta title="@core/Installing" />

# Core Components

<Anchor storyId="install">
  <Subheading>Installing</Subheading>
</Anchor>

```bash

#npm
npm install @axux/core

# yarn
yarn add @axux/core

```

<Anchor storyId="using">
  <Subheading>Using AxUX</Subheading>
</Anchor>

<cite type="code">src/index.tsx</cite>

```tsx
import { AxApplicationProvider, AxViewport } from "@axux/core";
import { Fragment, StrictMode } from "react";
import ReactDOM from "react-dom/client";
import { I18nextProvider } from "react-i18next";
import { Outlet, RouterProvider, createMemoryRouter } from "react-router";
import i18next from "./i18n";
import reportWebVitals from "./reportWebVitals";
import "./styles.css";

const root = ReactDOM.createRoot(document.getElementById("root"));

const router = createMemoryRouter([
  {
    path: "/",
    element: (
      <AxViewport>
        <div>Build the ui here</div>
        <Outlet />
      </AxViewport>
    ),
    children: [
      {
        path: "page",
        element: <Page />,
      },
      {
        path: "page",
        element: <Page />,
      },
    ],
  },
]);

const Wrapper = process.env.NODE_ENV === "development" ? Fragment : StrictMode;
root.render(
  <Wrapper>
    <AxApplicationProvider defaultPrimary="denim" defaultAccent="coral">
      <I18nextProvider i18n={i18next}>
        <RouterProvider router={router} />
      </I18nextProvider>
    </AxApplicationProvider>
  </Wrapper>,
);

// @ts-expect-error ignore
if (module.hot) {
  // @ts-expect-error ignore
  module.hot.accept();
}

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals(console.debug);
```
