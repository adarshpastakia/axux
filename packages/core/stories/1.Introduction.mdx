import {
  Title,
  Canvas,
  Story,
  Description,
  Heading,
  Subheading,
  Anchor,
} from "@storybook/addon-docs";
import { Meta } from "@storybook/blocks";
import {
  AxAnimation,
  AxApplicationProvider,
  AxAside,
  AxBreadcrumb,
  AxButton,
  AxCallout,
  AxCard,
  AxCollapsable,
  AxContent,
  AxDivider,
  AxErrorBoundary,
  AxFooter,
  AxHeader,
  AxFlexBox,
  AxLoader,
  AxIcon,
  AxMenu,
  AxMeter,
  AxSection,
  AxPage,
  AxProgress,
  AxTag,
  AxText,
  AxTheme,
  AxTitle,
  AxViewport,
  useApplicationContext,
} from "../src";

<Meta title="@core/Introduction" />

# Core Components

<Anchor storyId="app">
  <Subheading>Application</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxApplicationProvider</label>
    <Description of={AxApplicationProvider} />
  </div>
  <div className="comp-card">
    <label>AxViewport</label>
    <Description of={AxViewport} />
  </div>
  <div className="comp-card">
    <label>AxPage</label>
    <Description of={AxPage} />
  </div>
  <div className="comp-card">
    <label>AxErrorBoundary</label>
    <Description of={AxErrorBoundary} />
  </div>
</div>

<Anchor storyId="pageLayouts">
  <Subheading>Page Layout</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxSection</label>
    <Description of={AxSection} />
  </div>
  <div className="comp-card">
    <label>AxContent</label>
    <Description of={AxContent} />
  </div>
  <div className="comp-card">
    <label>AxAside</label>
    <Description of={AxAside} />
  </div>
  <div className="comp-card">
    <label>AxHeader</label>
    <Description of={AxHeader} />
  </div>
  <div className="comp-card">
    <label>AxFooter</label>
    <Description of={AxFooter} />
  </div>
</div>

<Anchor storyId="flexLayouts">
  <Subheading>Flex Layout</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxFlexBox</label>
    <Description of={AxFlexBox} />
  </div>
  <div className="comp-card">
    <label>AxFlexBox.Row</label>
    <Description of={AxFlexBox.Row} />
  </div>
  <div className="comp-card">
    <label>AxFlexBox.Col</label>
    <Description of={AxFlexBox.Col} />
  </div>
</div>

<Anchor storyId="typography">
  <Subheading>Typography</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxText</label>
    <Description of={AxText} />
  </div>
  <div className="comp-card">
    <label>AxText.Abbr</label>
    <Description of={AxText.Abbr} />
  </div>
  <div className="comp-card">
    <label>AxText.Mark</label>
    <Description of={AxText.Mark} />
  </div>
  <div className="comp-card">
    <label>AxText.Copy</label>
    <Description of={AxText.Copy} />
  </div>
  <div className="comp-card">
    <label>AxTitle</label>
    <Description of={AxTitle} />
  </div>
</div>

<Anchor storyId="buttons">
  <Subheading>Buttons</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxButton</label>
    <Description of={AxButton} />
  </div>
  <div className="comp-card">
    <label>AxButton.Group</label>
    <Description of={AxButton.Group} />
  </div>
  <div className="comp-card">
    <label>AxButton.Confirm</label>
    <Description of={AxButton.Confirm} />
  </div>
  <div className="comp-card">
    <label>AxButton.Action</label>
    <Description of={AxButton.Action} />
  </div>
  <div className="comp-card">
    <label>AxButton.Dropdown</label>
    <Description of={AxButton.Dropdown} />
  </div>
</div>
