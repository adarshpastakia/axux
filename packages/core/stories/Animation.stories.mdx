import {
  Anchor,
  ArgsTable,
  Canvas,
  Description,
  Heading,
  Meta,
  Story,
  Title,
} from "@storybook/addon-docs";
import { useState, useEffect } from "react";
import { AxAnimation, AxBadge, AxProgress } from "../src";

<Meta title="@core/Animations" />

<Title>Animations</Title>

<Heading storyId="core-animations--1a-check">Animated Circles</Heading>

<Canvas>
  <Story name="1a.Check">
    <AxAnimation.Check className="text-success-500" />
  </Story>
  <Story name="1b.Cross">
    <AxAnimation.Cross className="text-danger-500" />
  </Story>
  <Story name="1c.Info">
    <AxAnimation.Info className="text-primary-500" />
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxAnimation.Check} />

<Heading storyId="core-animations--2-a-bars">Animated Indicators</Heading>

<Canvas>
  <Story name="2a.Bars">
    <AxAnimation.Bars className="text-primary-500" />
  </Story>
  <Story name="2b.Spinner">
    <AxAnimation.Spinner className="text-4xl text-accent-500" />
  </Story>
  <Story name="2c.Card">
    <AxAnimation.Card className="w-64" showIcon />
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxAnimation.Bars} />

<Heading storyId="core-animations--3-a-progress-circle">Progress</Heading>

export const CircleTemplate = (args) => {
  const [value, setValue] = useState(24);
  useEffect(() => {
    const tmr = setInterval(() => {
      setValue(Math.random() * 100);
    }, 5000);
    return () => {
      clearInterval(tmr);
    };
  }, []);
  return <AxProgress.Circle size="md" value={value} />;
};
const CircleStory = CircleTemplate.bind({});
CircleStory.args = {};

export const BarTemplate = (args) => {
  const [value, setValue] = useState(24);
  useEffect(() => {
    const tmr = setInterval(() => {
      setValue(Math.random() * 100);
    }, 5000);
    return () => {
      clearInterval(tmr);
    };
  }, []);
  return (
    <div style={{ width: "40rem" }}>
      <AxProgress.Bar size="md" value={value} animate />
    </div>
  );
};
const BarStory = BarTemplate.bind({});
BarStory.args = {};

<Canvas>
  <Story name="3a.Progress Circle">{CircleStory}</Story>
  <Story name="3b.Progress Bar">{BarStory}</Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxProgress.Bar} />
