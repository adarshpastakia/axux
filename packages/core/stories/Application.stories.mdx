import {
  Anchor,
  ArgsTable,
  Canvas,
  Description,
  Heading,
  Meta,
  Story,
  Title,
} from "@storybook/addon-docs";
import { IconsStory } from "./application/Icons";
import { ViewportStory } from "./application/Viewport";
import { AxViewport, AxPage } from "../src";

export const LIPSUM =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus vehicula nulla luctus sem congue iaculis. Praesent lobortis gravida nibh non sodales. Ut lacinia nulla ac pellentesque ullamcorper. Phasellus non quam lorem. Etiam consectetur dapibus dapibus. Nunc diam libero, facilisis a rhoncus eget, fermentum non lacus. Quisque scelerisque hendrerit ipsum, et lobortis diam venenatis at. Sed facilisis lectus vitae nisl vestibulum ultrices. Vivamus luctus neque ut dolor blandit sodales.\n\nDonec maximus, dui laoreet maximus hendrerit, mauris mi varius ex, a dapibus est ex eget tellus. Sed nisl purus, cursus sed dictum a, egestas et turpis. Praesent scelerisque placerat cursus. Phasellus et dui vel velit faucibus venenatis eu mattis ante. Maecenas eget vulputate dolor. Duis eu lobortis nisl. Vestibulum ullamcorper tristique rutrum. Mauris non est et neque cursus lobortis eu sed neque. Duis risus erat, malesuada eu nisi in, efficitur egestas eros.\n\nDonec scelerisque sit amet dolor in pulvinar. Integer id dolor lorem. Morbi dignissim lorem id tincidunt porttitor. Curabitur feugiat, dui et varius finibus, nisi ante rutrum elit, ac venenatis libero massa nec elit. Vivamus tincidunt libero quis nulla lobortis aliquet. Vestibulum felis nisl, ultricies ut lacus in, viverra elementum justo. Proin ut nisi enim.";

<Meta title="@core/Application" />

<Title>Viewport</Title>

<Heading storyId="core-application--1-viewport">Application Viewport</Heading>

<Canvas>
  <Story name="1.Viewport" height="32rem" parameters={{ layout: "none" }}>
    {ViewportStory.bind({})}
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" components={{ AxViewport, AxPage }} />

<Canvas>
  <Story name="Icons" height="32rem">
    {IconsStory.bind({})}
  </Story>
</Canvas>
