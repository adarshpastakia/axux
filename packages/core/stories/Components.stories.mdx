import {
  Anchor,
  ArgsTable,
  Canvas,
  Description,
  Heading,
  Meta,
  Story,
  Title,
} from "@storybook/addon-docs";
import { useState, useEffect } from "react";
import {
  AxDivider,
  AxLoader,
  AxAnimation,
  AxFlexBox,
  AxMeter,
  AxCallout,
  AxText,
  AxCard,
} from "../src";
import { CalloutProps } from "../src/components/Callout";
import { CalloutStory, CalloutDangerStory } from "./components/CalloutStory";

export const LIPSUM =
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus vehicula nulla luctus sem congue iaculis. Praesent lobortis gravida nibh non sodales. Ut lacinia nulla ac pellentesque ullamcorper. Phasellus non quam lorem. Etiam consectetur dapibus dapibus. Nunc diam libero, facilisis a rhoncus eget, fermentum non lacus.";

<Meta title="@core/Page Components" />

<Title>Page Components</Title>

<Heading storyId="core-page-components--1-divider">Divider</Heading>

<Canvas>
  <Story name="1.Divider" args={{ className: "" }}>
    {(args) => (
      <div style={{ width: 600 }}>
        <AxDivider {...args} />
      </div>
    )}
  </Story>
</Canvas>

<Heading storyId="core-page-components--2-divider-label">Divider Label</Heading>

<Canvas>
  <Story
    name="2.Divider Label"
    args={{
      className: "",
      color: "",
      align: "center",
      applyBg: false,
      width: 1,
    }}
  >
    {(args) => (
      <div style={{ width: 600 }}>
        <AxDivider {...args}>Divider label</AxDivider>
      </div>
    )}
  </Story>
</Canvas>

<Heading storyId="core-page-components--3-rainbow-divider">
  Rainbow Divider
</Heading>

<Canvas>
  <Story
    name="3.Rainbow Divider"
    args={{ className: "", align: "center", rainbow: true, children: "" }}
  >
    {(args) => (
      <div style={{ width: 600 }}>
        <AxDivider {...args} />
      </div>
    )}
  </Story>
</Canvas>

<Heading storyId="core-page-components--4-vertical-divider">
  Vertical Divider
</Heading>

<Canvas>
  <Story name="4.Vertical Divider" args={{ className: "" }}>
    {(args) => (
      <div style={{ height: 200 }} className="flex justify-center">
        <AxDivider {...args} vertical />
      </div>
    )}
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxDivider} />

<Heading storyId="core-page-components--5-loader-overlay">
  Loader Overlay
</Heading>

<Canvas>
  <Story
    name="5.Loader Overlay"
    args={{
      className: "text-accent",
      label: "",
      loaderClass: "text-6xl text-accent",
      strategy: "absolute",
    }}
  >
    {({ strategy, className, label, loaderClass }) => (
      <div className="aspect-video relative m-auto" style={{ width: 600 }}>
        <AxLoader strategy={strategy} className={className}>
          <AxAnimation.Bars className={loaderClass} />
          {label && <div>{label}</div>}
        </AxLoader>
      </div>
    )}
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxLoader} />

<Heading storyId="core-page-components--6-a-callout">Callout</Heading>

<Canvas>
  <Story name="6a.Callout" width="24rem">
    {CalloutStory}
  </Story>
  <Story name="6b.Callout" width="24rem">
    {CalloutDangerStory}
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxCallout} />

<Heading storyId="core-page-components--7-meter">Meter</Heading>

<Canvas>
  <Story
    name="7.Meter"
    args={{ className: "", size: "sm", value1: 92, value2: 56, value3: 72 }}
  >
    {({ value1, value2, value3, ...args }) => (
      <AxFlexBox>
        <AxFlexBox.Row>
          <AxFlexBox.Col width={180}>
            <AxMeter {...args} value={value1} color={undefined} />
          </AxFlexBox.Col>
          <AxFlexBox.Col width={180}>
            <AxMeter {...args} value={value2} color="primary" />
          </AxFlexBox.Col>
          <AxFlexBox.Col width={180}>
            <AxMeter {...args} value={value3} color="#1d4ed8" />
          </AxFlexBox.Col>
        </AxFlexBox.Row>
      </AxFlexBox>
    )}
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxMeter} />

<Heading storyId="core-page-components--8-card">Card</Heading>

<Canvas>
  <Story
    name="8.Card"
    args={{ className: "p-4", isPlain: false, isActive: false }}
  >
    {(args) => (
      <AxFlexBox>
        <AxFlexBox.Row>
          <AxFlexBox.Col width="24rem">
            <AxCard {...args}>
              <AxText>
                {LIPSUM}
                <AxText.Copy text={LIPSUM} />
              </AxText>
            </AxCard>
          </AxFlexBox.Col>
          <AxFlexBox.Col width="24rem">
            <AxCard {...args} onClick={() => undefined}>
              <AxText>{LIPSUM}</AxText>
              <AxText.Badge color="danger" placement="top-end">
                new
              </AxText.Badge>
            </AxCard>
          </AxFlexBox.Col>
        </AxFlexBox.Row>
      </AxFlexBox>
    )}
  </Story>
</Canvas>

<ArgsTable sort="requiredFirst" of={AxCard} />
