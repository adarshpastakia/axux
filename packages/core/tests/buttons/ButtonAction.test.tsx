/**
 * AxUX React UI Framework with Tailwind CSS
 * @author    : Adarsh Pastakia
 * @version   : 4.0.0
 * @copyright : 2024
 * @license   : MIT
 */

import { composeStories } from "@storybook/testing-react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import React from "react";
import * as stories from "../../stories/buttons/ButtonAction.stories";

const { Action } = composeStories(stories);

describe("Action Button", () => {
  it("should render", () => {
    const fn = jest.fn(() => true);
    const fragment = render(
      <Action message="Done message" onClick={fn}>
        Click Me
      </Action>,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render animation on true result", async () => {
    const fn = jest.fn(() => true);
    const fragment = render(
      <Action message="Done message" onClick={fn}>
        Click Me
      </Action>,
    );
    fireEvent.click(
      fragment.container.querySelector(".ax-button__inner") as HTMLElement,
    );
    await waitFor(() => {
      expect(fragment.container.querySelector(".animated-svg")).not.toBeNull();
      expect(
        fragment.container.querySelector(".ax-popover.ax-tooltip"),
      ).not.toBeNull();
    });
    fragment.unmount();
  });

  it("should not render animation on false result", () => {
    const fn = jest.fn(() => false);
    const fragment = render(
      <Action message="Done message" onClick={fn}>
        Click Me
      </Action>,
    );
    fireEvent.click(
      fragment.container.querySelector(".ax-button__inner") as HTMLElement,
    );
    expect(fragment.container.querySelector(".animated-svg")).toBeNull();
    expect(
      fragment.container.querySelector(".ax-popover.ax-tooltip"),
    ).toBeNull();
    fragment.unmount();
  });

  it("should be disabled", () => {
    const fn = jest.fn();
    const fragment = render(
      <Action isDisabled onClick={fn}>
        Click Me
      </Action>,
    );
    fireEvent.click(
      fragment.container.querySelector(".ax-button__inner") as HTMLElement,
    );
    expect(fragment.container.querySelector(".animated-svg")).toBeNull();
    expect(
      fragment.container.querySelector(".ax-popover.ax-tooltip"),
    ).toBeNull();
    fragment.unmount();
  });
});
