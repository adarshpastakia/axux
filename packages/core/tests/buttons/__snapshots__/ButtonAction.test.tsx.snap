// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button should render 1`] = `"<div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div data-active="false" data-round="false" data-square="false" data-spinner="false" data-loading="false" data-disabled="false" class="ax-button "><button data-active="false" data-action-done="false" role="button" class="ax-button__inner" data-disabled="false" tabindex="0" type="button"><div class="ax-button__label">Click Me</div></button></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;
